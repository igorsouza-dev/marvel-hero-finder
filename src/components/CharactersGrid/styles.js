import styled from 'styled-components';
import { device, screenSize } from 'styles/variables';

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(6, ${(props) => props.size}px);
  grid-gap: 10px;
  padding: 10px;
  @media only screen and (max-width: ${screenSize.mobileL}) {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    max-width: ${screenSize.mobileL};
    overflow-x: scroll;
  }
  @media only screen and (${device.mobileL}) {
    grid-template-columns: repeat(2, ${(props) => props.size}px);
  }
  @media only screen and (${device.tablet}) {
    grid-template-columns: repeat(3, ${(props) => props.size}px);
  }
  @media only screen and (${device.laptop}) {
    grid-template-columns: repeat(5, ${(props) => props.size}px);
  }
`;
