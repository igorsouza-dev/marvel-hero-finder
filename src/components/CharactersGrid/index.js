import React from 'react';
import PropTypes from 'prop-types';

import CharacterCard from 'components/CharacterCard';
import { Grid } from './styles';

function CharactersGrid({ characters }) {
  return (
    <>
      <Grid>
        {characters.map((character) => (
          <CharacterCard key={character.id} character={character} />
        ))}
      </Grid>
    </>
  );
}

CharactersGrid.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};


export default CharactersGrid;
