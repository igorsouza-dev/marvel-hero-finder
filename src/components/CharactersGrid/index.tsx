import React from 'react';
import PropTypes from 'prop-types';

import CharacterCard from 'components/CharacterCard';
import Character from 'types/Character';
import { Grid, Content, InfoText } from './styles';

interface CharactersGridProps {
  characters: Character[];
}

const CharactersGrid: React.FC<CharactersGridProps> = ({ characters }) => (
  <>
    {characters.length ? (
      <Grid>
        {characters.map((character: Character) => (
          <CharacterCard key={character.id} character={character} />
        ))}
      </Grid>
    ) : (
      <Content>
        <InfoText>No characters found.</InfoText>
      </Content>
    )}
  </>
);

CharactersGrid.propTypes = {
  characters: PropTypes.array.isRequired,
};

export default CharactersGrid;
