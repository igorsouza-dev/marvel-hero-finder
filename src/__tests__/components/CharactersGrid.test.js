import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import reducer, { INITIAL_STATE } from 'store/reducers';
import CharactersGrid from 'components/CharactersGrid';

function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, INITIAL_STATE) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}


describe('<CharactersGrid/>', () => {
  afterEach(cleanup);
  test('renders a grid of characters cards', () => {
    const character = {
      id: '123456',
      name: 'Iron Man',
      thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
        extension: 'jpg',
      },
    };
    const history = createMemoryHistory();
    const { container, getByText } = renderWithRedux(
      <Router history={history}>
        <CharactersGrid characters={[character]} />
      </Router>,
    );
    const linkElement = getByText(/Iron Man/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
