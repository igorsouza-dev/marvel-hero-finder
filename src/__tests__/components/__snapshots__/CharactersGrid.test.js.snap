// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharactersGrid/> renders a grid of characters cards 1`] = `
<div
  class="sc-bwzfXH hTrUmU"
>
  <div
    class="sc-bdVaJa dKPgmu"
  >
    <div>
      <strong>
        Iron Man
      </strong>
    </div>
  </div>
</div>
`;
